search/search_param_types.go

```diff
@@ -906,6 +968,15 @@ func (r *ReferenceParam) setInfo(info SearchParamInfo) {
 
 func (r *ReferenceParam) getQueryParamAndValue() (string, string) {
 	switch t := r.Reference.(type) {
+	case ReverseChainedQueryReference:
+		searchParams := t.Query.Params()
+		if len(searchParams) != 1 {
+			panic(createInternalServerError("MSG_PARAM_CHAINED", "Unknown chained parameter name \"\""))
+		}
+		// e.g. "gender=male"
+		qParam, qValue := searchParams[0].getQueryParamAndValue()
+		// (e.g. "_has:Observation:subject:gender", "male")
```

Comment:
These examples aren't correct.  Observations don't have gender.  That last part has to be a search param that applies to _Observation_.  The FHIR spec uses this example:
```
/Patient?_has:Observation:patient:code=1234-5
```